---
title: "firstdogma"
output: rmarkdown::pdf_document
vignette: >
  %\VignetteIndexEntry{firstdogma}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

Link to gitHub repository:

https://github.com/rforbiodatascience25/group_15_package/tree/master

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

Load the firstdogma library to access the five functions made in class

```{r setup}
library("firstdogma")
```

## Function 1

This function generates a random DNA sequence. This is how to use it

```{r}
generate_dna_sequence(50)
```

It will give randomly generated DNA sequence with a length of 50 bases.

## Function 2

This is how you use function T_to_U

```{r}
T_to_U("TATATA")
```

As you can see the function substitutes all Ts to Us

## Function 3

The split_codons function allows you to split a nucleotide sequence into codons

```{r split_codons function}

split_codons("ATGCGTACG")
```

If you want to start somewhere else use the start argument like below

```{r}
split_codons("ATGCGTACG", start = 2)
```

## Function 4

This function translates the character vector of codons into respective amino acids based on codon_table defined previously. This is the translation phase of the central dogma.

```{r}
translation(codon_table)
```

## Function 5
`AA_counts_plot` takes a sequence of amino acid one letter codes as a string and generates a bargraph of the counts for each amino acid. It also works with DNA or RNA sequences. Any characters which do not correspond to an amino acid one letter code or DNA/RNA will be ignored.

To use, simply call it with a string of amino acids or DNA/RNA:

```{r AA_counts_plot_example}
#Amino acids sequence
AA_counts_plot("SSFPTAYYHQNNQQ")

#DNA
AA_counts_plot("GGTACCTAGGATCA")
```

